pkg/healthcheck/health_controller.go

```diff
@@ -156,27 +162,25 @@ func (hc *HealthController) RunServer(stopCh <-chan struct{}, wg *sync.WaitGroup
 		hc.HTTPEnabled = false
 	}
 
-	select {
-	case <-stopCh:
-		glog.Infof("Shutting down health controller")
-		if hc.HTTPEnabled {
-			if err := srv.Shutdown(context.Background()); err != nil {
-				glog.Errorf("could not shutdown: %v", err)
-			}
+	// block until we receive a shut down signal
+	<-stopCh
+	glog.Infof("Shutting down health controller")
+	if hc.HTTPEnabled {
+		if err := srv.Shutdown(context.Background()); err != nil {
+			glog.Errorf("could not shutdown: %v", err)
 		}
-		return nil
 	}
 }
 
 //RunCheck starts the HealthController's check
-func (hc *HealthController) RunCheck(healthChan <-chan *ControllerHeartbeat, stopCh <-chan struct{}, wg *sync.WaitGroup) error {
+func (hc *HealthController) RunCheck(healthChan <-chan *ControllerHeartbeat, stopCh <-chan struct{}, wg *sync.WaitGroup) {
```

Comment:
Removed error return value here since it was not used and it's a go routine, so probably an error channel makes more sense.